app/controllers/lessons/previews_controller.rb

```diff
@@ -1,8 +1,20 @@
 module Lessons
   class PreviewsController < ApplicationController
-    def show; end
+    def show
+      @preview = LessonPreview.find_or_initialize_by(id: params[:uuid])
+    end
 
     def create
+      preview_link = LessonPreview.create(lesson_preview_params)
```

Comment:
I don't think the error branch will ever be called with this since create will always return the LessonPreview object. We'll need to do something like this instead:

```ruby
    def create
      preview_link = LessonPreview.new(lesson_preview_params)
        
      if preview_link.save
        render json: { preview_link: preview_link(preview_link.id) }, status: :created
      else
        render json: { errors: preview_link.errors.full_messages }, status: :unprocessable_entity
      end
    end
```